src/test/java/org/mockitousage/verification/DelayedExecution.java

```diff
@@ -1,54 +1,78 @@
 package org.mockitousage.verification;
 
-import org.mockitousage.IMethods;
+import static java.lang.System.currentTimeMillis;
+import static java.lang.Thread.MAX_PRIORITY;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.concurrent.locks.LockSupport.parkUntil;
 
-import java.util.ArrayList;
-import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.TimeUnit;
 
 class DelayedExecution {
+    private static final int CORE_POOL_SIZE = 3;
+    /**
+     * Defines the number of milliseconds we expecting a Thread might need to unpark, we use this to avoid "oversleeping" while awaiting the deadline for  
+     */
+    private static final long MAX_EXPECTED_OVERSLEEP_MILLIS = 50;
     private final ScheduledExecutorService executor;
-    private final IMethods mock;
-    private final int delay;
-    private final TimeUnit timeUnit;
-    private final ArrayList<Character> invocations = new ArrayList<Character>();
-
-    DelayedExecution(ScheduledExecutorService executor, IMethods mock, int delay, TimeUnit timeUnit) {
-        this.executor = executor;
-        this.mock = mock;
-        this.delay = delay;
-        this.timeUnit = timeUnit;
+
+    public DelayedExecution() {
+        this.executor = newScheduledThreadPool(CORE_POOL_SIZE, maxPrioThreadFactory());
     }
 
-    void recordAsyncCall(char c) {
-        invocations.add(c);
+    public void callAsync(long delay, TimeUnit timeUnit, Runnable r) {
+        long deadline = timeUnit.toMillis(delay) + currentTimeMillis();
```

Comment:
I think the code should check the deadline is not too small, especially since the code will decrease this amount [here](https://github.com/mockito/mockito/pull/704/files#diff-305ace128d183213cab9933b42729f20R44). The best option imho is to perform the deadline calculation in `delayedExecution`